src/openassetio-core-c/include/openassetio/c/errors.h

```diff
@@ -14,6 +14,16 @@ extern "C" {
 
 /// Error code indicating an OK result from a C API function.
 const int OPENASSETIO_NS(kOK) = 0;
```

Comment:
Any reason this is not just a simple enum?
(I imagine automated tools for language bindings may be able to gain more information from an enum than just a list of constants.)

Reply:
Good point, it's also better start out restrictive.  Using an enum is a bit more verbose in the C API function signatures (i.e. return type is an enum type rather than simply `int`), but that's a minor point.

Reply:
As part of addressing https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/330#discussion_r855512524 I rewrote the git history, where I also updated the error codes to use an `enum`.